x-pack/filebeat/input/awscloudwatch/input.go

```diff
@@ -91,15 +93,24 @@ func (in *cloudwatchInput) Run(inputContext v2.Context, pipeline beat.Pipeline)
 	ctx := v2.GoContextFromCanceler(inputContext.Cancelation)
 	log := inputContext.Logger
 
+	// setup status reporter
+	in.status = newCWStateReporter(inputContext, log)
+
+	defer in.status.UpdateStatus(status.Stopped, "Input stopped")
+	in.status.UpdateStatus(status.Starting, "Input starting")
+
 	handler, err := newStateHandler(log, in.config, in.store)
 	if err != nil {
+		in.status.UpdateStatus(status.Failed, "State registry creation failure")
```

Comment:
We've typically included the error with the status message, so the user has an idea of what exactly went wrong.

See for example: http://github.com/elastic/beats/blob/6409d005a31e0147a40f0872794880b3df15a69c/x-pack/filebeat/input/cel/input.go#L122

Same would apply of other degraded and error scenarios.

Reply:
@taylor-swanson thanks for this. I have added missing error messages for degrading scenarios with commit https://github.com/elastic/beats/pull/45679/commits/eb86641967ff90e4bcb52afd1740726d4a6e69ef